### PR TITLE
ClusterRunner init.d service scripts

### PR DESCRIPTION
**Pull request by tjlee0909**
This is the init.d script that Box uses internally for ClusterRunner deployment on our RedHat linux platform.

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @tjlee0909 has signed the CLA. Thanks for the pull request!

File: clusterrunner_ha [hunk 1]
```diff
@@ -0,0 +1,309 @@
...
+
+    # Build ssh/scp command
+    SSH_COMMAND="ssh -o ConnectTimeout=5 -o BatchMode=yes -o PasswordAuthentication=no -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no $TARGET \"$COMMAND\""
+    if $IS_ASYNC
+    then
+        SSH_COMMAND="$SSH_COMMAND &"
+    fi
+
+    # Execute remote command as jenkins
+    sudo su jenkins -c "$SSH_COMMAND"
```
Comment (tjlee0909) [inline clusterrunner_ha]: Obviously, replace 'jenkins' with whatever user you are using to run ClusterRunner.

File: clusterrunner_ha.sysconfig [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+#
+SLAVE_STARTING_PORT="43001"
+
+## Type:        string
+## ServiceRestart: clusterrunner_ha
+#
+# The path to where the clusterrunner.conf file lives on the
+# current host.
+#
+CLUSTERRUNNER_CONF_PATH="/home/jenkins/.clusterrunner/clusterrunner.conf"
```
Comment (tjlee0909) [inline clusterrunner_ha.sysconfig]: Remember to replace this path with wherever you have deployed you clusterrunner.conf file.

File: clusterrunner_ha.sysconfig [hunk 3]
```diff
@@ -0,0 +1,70 @@
+## Type:        string
+## Default:     "cluster-master.dev.box.net"
+## ServiceRestart: clusterrunner_ha
+#
+# The hostname of the proxy for the Cluster Master service.
+#
+CLUSTERMASTER_PROXY_HOSTNAME="master_host.domain.net"
```
Comment (tjlee0909) [inline clusterrunner_ha.sysconfig]: At Box, we implemented high-availability for ClusterRunner via and nginx proxy. So, for non-HA deployments, you can just set this URL to the same as your cluster-master host's url (MASTER_HOSTS_CSV).

File: clusterrunner_ha.sysconfig [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+## Type:        string
+## ServiceRestart: clusterrunner_ha
+#
+# A comma-separated list of hostnames that will run the ClusterRunner
+# master service.
+#
+MASTER_HOSTS_CSV="master_host.domain.met"
```
Comment (tjlee0909) [inline clusterrunner_ha.sysconfig]: Because you likely won't be deploying ClusterRunner initially in HA-mode, you should just specify one hostname here.

**Comment by tjlee0909** [comment]
Note: you may have to fiddle a little bit with the config and the script, as we made some assumptions that are environment specific in this script.

**Comment by tjlee0909** [comment]
1). Place the clusterrunner_ha script in /etc/init.d/clusterrunner_ha, and the clusterrunner_ha.sysconfig in /etc/sysconfig/clusterrunner_ha.

2). Update the sysconfig parameters.

After that's done, simply do: "sudo service clusterrunner_ha start" in order to deploy, and "sudo service clusterrunner_ha stop" to kill the service.

File: clusterrunner_ha.sysconfig [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+# current host.
+#
+CLUSTERRUNNER_CONF_PATH="/home/jenkins/.clusterrunner/clusterrunner.conf"
+
+## Type:        string
+## ServiceRestart: clusterrunner_ha
+#
+# The URL to the linux tarball of ClusterRunner binaries.
+#
+CLUSTERRUNNER_BINARY_URL="https://cloud.box.com/shared/static/2pl4pi6ykvrbb9d06t4m.tgz"
```
Comment (josephharrington) [inline clusterrunner_ha.sysconfig]: @tjlee0909 is this the same link as the publicly available linux package or is it something internal?

Comment (tjlee0909) [inline clusterrunner_ha.sysconfig]: That's the same publicly available link. Internally we point to https://jenkins/crlinuxbuild/latestSuccessful

**Pull request closed without merging**